Issue/1188

dogeared opened the pull request:
Resolves #1188 
Resolves #1008 

To review, please don't just look at the Pull Request side-by-side diff. The docs need to be reviewed in rendered form to get a feel for what a customer sees (to see if it makes sense and is usable).

To review:

Check out the pull request as a new branch:

```
$ git fetch origin pull/1188/head:pr-1188
```

Then switch to this new branch:

```
$ git checkout pr-1188
```

Then generate and open/read the docs:

```
$ cd docs
$ ./build.sh spring # you shouldn't see any errors or warnings here
$ open build/spring/build/html/index.html
```

The complete set of integration are also staged here: http://stormpath-docs-stage.herokuapp.com/

Pay particular attention to the tutorial sections:

* [Spring Tutorial](http://stormpath-docs-stage.herokuapp.com/spring/tutorial.html)
* [Spring Boot Tutorial](http://stormpath-docs-stage.herokuapp.com/springboot/tutorial.html)

The Spring Security sections of the `Spring Boot Tutorial` should *not* have any references to `.apply(stormpath())` while the same sections of the `Spring Tutorial` should include references to `.apply(stormpath())`.

mrioan commented:
@dogeared in Quickstart -> Next Steps -> There is a reference to Spring Boot : `Dig in to our examples to see more Stormpath Spring Boot in action.`

mrioan commented on docs/source/conf.py:
```diff
@@ -77,9 +81,12 @@
     project = u'Stormpath Default Spring Boot Starter'
     apptype = u'application'
+elif tags.has('spring'):
+    project = u'Stormpath Spring Security WebMVC Integration'
```
@dogeared I am not sure about referring to Spring Security in the tile of this Spring Documentation. Spring Boot documentation is titled `Stormpath Default Spring Boot Starter`. It does not have a explicit mention of Spring Security. I think we should call it `Stormpath Spring WebMVC Integration`. Spring Security will be mentioned internally but not in the title. I am open to discussion, I think Spring Security should not be in the title but I am totally fine is the rest of the team feels otherwise.

dogeared replied:
makes sense

mrioan commented on docs/source/quickstart.rst:
```diff
@@ -140,12 +151,18 @@ works correctly, you will need a Spring Security configuration class and apply t
 Without this, you will see a browser popup prompting for authentication, which is the default basic authentication behavior for Spring Security.
 
-Also, by default, all paths are locked down with Spring Security. Stormpath's Spring Security integration follows this idiomatic behavior.
+#elseif( $springboot )
```
you have `springboot` in the `if` statement, this `elseif` block will never be executed

dogeared replied:
There's an outer if and an inner one. It's a little hard to see because the lack of indentation:

```
#if( $spring or $springboot or $sczuul )
    ...
#if( $spring )
    ...
#elseif( $springboot )
    ...
#end
    ...
#end
```

mrioan replied:
oh, good, thanks for the clarification

mrioan commented on docs/source/quickstart.rst:
```diff
@@ -290,14 +308,14 @@ Did you experience any problems with this quickstart?  It might not have worked
              <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
              <java.version>1.8</java.version>
-             <spring-security.version>4.1.2.RELEASE</spring-security.version>
+             <spring-security.version>4.1.3.RELEASE</spring-security.version>
```
We will be using `4.2.0.RELEASE` in Java SDK 1.3.0

mrioan commented:
@dogeared I see you assigned it back to me but, unless I am seeing wrongly, my comments were not addressed. Back to you.

dogeared commented:
You are correct! Working on your comments now. Thanks!

dogeared commented:
back to you @mrioan

mrioan commented on docs/source/conf.py:
```diff
@@ -151,11 +158,17 @@
     """
+elif tags.has('spring'):
+    rst_prolog = """
+    .. |project| replace:: Stormpath Spring Security WebMVC Integration
```
For consistency with the new title the project should also be updated to `Stormpath Spring WebMVC Integration`

mrioan commented on docs/source/conf.py:
```diff
@@ -249,8 +262,10 @@
+elif tags.has('springboot'):
     htmlhelp_basename = 'StormpathDefaultSpringBootStarterDoc'
+elif tags.has('spring'):
+    htmlhelp_basename = 'StormpathSpringSecurityWebMVCDoc'
```
for consistency with the new name this based name should be `StormpathSpringWebMVCIntegrationDoc `

mrioan commented on docs/source/conf.py:
```diff
@@ -281,11 +296,17 @@
     ]
+elif tags.has('spring'):
+    latex_documents = [
+        ('index', 'StormpathSpringSecurityWebMVC.tex', u'Stormpath Spring Security WebMVC Integration Documentation',
```
Idem

mrioan commented on docs/source/conf.py:
```diff
@@ -344,12 +365,18 @@
     ]
+elif tags.has('spring'):
+    texinfo_documents = [
+        (master_doc, 'StormpathSpringSecurityWebMVC', u'Stormpath Spring Security WebMVC Integration Documentation',
```
Idem

mrioan commented:
Two additional comments @dogeared:

1. you should add spring to `docs/readme.md`.
2. I see this comment was not addressed: `In the tutorial page do a search for "Boot". You will see 4 matches there.`

dogeared commented:
@mrioan - one of the four occurrences of the word "Boot" was appropriate for the context. I've addressed the other three.

And, I updated the docs/readme.

Thanks!

mrioan commented:
LGTM. Feel free to merge!
